### PR TITLE
planner: fix "can't find column" when projection wrongly added above table reader after agg pushed down (#23804)

### LINKED ISSUE #23736: Can't find column in schema Column: [Column#10] Unique key: [] when using stream_agg()  hint
## Bug Report

Please answer these questions before submitting your issue. Thanks!

### 1. Minimal reproduce step (Required)
```sql
create table t0(a int, b int, c int as (a + b) virtual, unique index (c) invisible);
create table t1(a int, b int, c int as (a + b) virtual);
insert into t0(a, b) values (12, -1), (8, 7);
insert into t1(a, b) values (12, -1), (8, 7);
select /*+ stream_agg() */ count(1) from t0 where c > 10 and b < 2;
select /*+ stream_agg() */ count(1) from t1 where c > 10 and b < 2;
```

### 2. What did you expect to see? (Required)
```sql
mysql > select /*+ stream_agg() */ count(1) from t0 where c > 10 and b < 2;
+----------+
| count(1) |
+----------+
| 1        |
+----------+

1 row in set
Time: 0.133s
mysql > select /*+ stream_agg() */ count(1) from t1 where c > 10 and b < 2;
+----------+
| count(1) |
+----------+
| 1        |
+----------+

```
### 3. What did you see instead (Required)
```sql
mysql > select /*+ stream_agg() */ count(1) from t0 where c > 10 and b < 2;
(1105, "Can't find column hchwang.t0.b in schema Column: [Column#6] Unique key: []")
mysql > select /*+ stream_agg() */ count(1) from t1 where c > 10 and b < 2;
(1105, "Can't find column hchwang.t1.b in schema Column: [Column#6] Unique key: []")
```
### 4. What is your TiDB version? (Required)
master: f43f750fb551f2f

### PR DESCRIPTION
**Pull request by ti-srebot**
cherry-pick #23804 to release-5.0
You can switch your code base to this Pull Request by using [git-extras](https://github.com/tj/git-extras):
```bash
# In tidb repo:
git pr https://github.com/pingcap/tidb/pull/23816
```

After apply modifications, you can push your change to this PR via:
```bash
git push git@github.com:ti-srebot/tidb.git pr/23816:release-5.0-008b91b12b07
```

---

### What problem does this PR solve?

Issue Number: close #23736 

Problem Summary:

A projection was added between the partial agg and the final agg, and the schema between them was broken.

### What is changed and how it works?

Not to add the extra projection if the agg was pushed down.

### Related changes

- Need to cherry-pick to the release branch

### Check List 

Tests 

- Unit test
- Integration test

### Release note 

- no release note.

### CODE REVIEW

File: planner/core/integration_test.go [hunk 1]
```diff
@@ -3033,3 +3033,125 @@ func (s *testIntegrationSuite) TestIndexMergeTableFilter(c *C) {
 		"10 1 1 10",
 	))
 }
+<<<<<<< HEAD
```
Comment (zimulala) [inline planner/core/integration_test.go:3036]: @time-and-fate 
Please fix this conflict.

**Review by winoros — APPROVED** [review]

**Pull request merged**